### PR TITLE
Update Android minSDK and Kotlin version to adopt Jetpack Compose [MOB-18949]

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description

UIServices provided by `aepsdk-core-android` will now move to Jetpack Compose based UI. To start with this migration uplevel current language and gradle dependencies to ones that support Compose. [Jetpack needs Kotlin 1.5+](https://developer.android.com/jetpack/androidx/releases/compose-kotlin#pre-release_kotlin_compatibility) and is supported on [minSdk versions 21+](https://developer.android.com/jetpack/compose/setup#create-new)
- Change `minSdk` requirement from `API 19` to `API 21`.
- `Android Gradle Plugin 7.3.1` needs at least `Kotlin 1.5.21`. However, `1.5.21` results in incompatibility issues with Android Studio Flamingo onwards. So use `1.5.31` instead.
- Use the mapping between Compose Compiler version and Kotlin version [here](https://developer.android.com/jetpack/androidx/releases/compose-kotlin#pre-release_kotlin_compatibility) and use `kotlinCompilerExtensionVersion 1.0.5` and Compose version 1.0.5
- It is recommended that Kotlin gradle plugin version matches the Kotlin version being used. So down level Kotlin gradle plugin version from `1.7.20` to `1.5.31`
- Add `compose.runtime` `compose.material` and `activity-compose` dependancies to Core for implementation.
- Remove explicit dependency on `org.jetbrains.kotlinx:kotlinx-coroutines-core` for coroutines as they are included with compose libraries.
- To keep the Kotlin gradle version same between the root project and `testapp-kotlin` and `integration-tests` modules, use compose `runtime` , `ui` and `core:ktx` dependancies that are compatible with Compose 1.0.5
- Update the gradle script to add POM dependancies
- Standardize the minSDK, targetSDK, compileSDK versions being used across projects in this repo.

A note on the `public static final field $stable I` api dump update: Looks like this is due to Compose compiler to marking some classes (final objects and final data classes) with this  internally. I am trying to check if there is a way for this to be prevented. 

## Related Issue
Internal MOB-18949

## Motivation and Context
UIServices provided by `aepsdk-core-android` will now move to Jetpack Compose based UI. To start with this migration uplevel current language and gradle dependencies to ones that support Compose.

## How Has This Been Tested?
- Unit tests
- Integration tests
- Functional tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by yangyansong-adbe** [comment]
@prudrabhat Would it be possible to utilize BOM to manage compose dependencies? https://developer.android.com/jetpack/compose/bom

**Comment by prudrabhat** [comment]
> @prudrabhat Would it be possible to utilize BOM to manage compose dependencies? https://developer.android.com/jetpack/compose/bom

I also wanted to use BOM if possible to eliminate version tracking but, Compose BOM is only available starting Compose versions 1.3.0 (that needs Kotlin 1.7). So I had to individual versions itslef.

File: code/build.gradle [hunk 1]
```diff
@@ -39,11 +40,11 @@ allprojects {
 }
 
 ext {
-    minSdkVersion = 19
+    minSdkVersion = 21
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/build.gradle:43]: The latest Android Studio Stable release is Giraffe which supports max AGP 8.1. When I attempted to use this and uplevel the targetSDK to 34, I notice that there are still warnings and build errors. It looks like we will need to wait for the next AGP release to update to 34 stably. 

```
We recommend using a newer Android Gradle plugin to use compileSdk = 34

This Android Gradle plugin (8.1.0) was tested up to compileSdk = 33 (and compileSdkPreview = "UpsideDownCakePrivacySandbox").

You are strongly encouraged to update your project to use a newer
Android Gradle plugin that has been tested with compileSdk = 34.

If you are already using the latest version of the Android Gradle plugin,
you may need to wait until a newer version with support for compileSdk = 34 is available.

```

File: code/core/build.gradle [hunk 2]
```diff
@@ -111,6 +117,16 @@ dependencies {
     implementation "androidx.cardview:cardview:1.0.0"
     //noinspection GradleDependency
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+
+    // Jetpack Compose runtime
+    implementation "androidx.compose.runtime:runtime:${compose_version}"
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/build.gradle:122]: Done! Added some groovy code to add dependancies to POM file.

**Pull request merged**